src/HtmlUtils.js

```diff
@@ -71,73 +67,27 @@ export function containsEmoji(str) {
     return SURROGATE_PAIR_PATTERN.test(str) || SYMBOL_PATTERN.test(str);
 }
 
-/* modified from https://github.com/Ranks/emojione/blob/master/lib/js/emojione.js
- * because we want to include emoji shortnames in title text
- */
-function unicodeToImage(str, addAlt) {
-    if (addAlt === undefined) addAlt = true;
-
-    let replaceWith; let unicode; let short; let fname;
-    const mappedUnicode = emojione.mapUnicodeToShort();
-
-    str = str.replace(emojione.regUnicode, function(unicodeChar) {
-        if ( (typeof unicodeChar === 'undefined') || (unicodeChar === '') || (!(unicodeChar in emojione.jsEscapeMap)) ) {
-            // if the unicodeChar doesnt exist just return the entire match
-            return unicodeChar;
-        } else {
-            // get the unicode codepoint from the actual char
-            unicode = emojione.jsEscapeMap[unicodeChar];
-
-            short = mappedUnicode[unicode];
-            fname = emojione.emojioneList[short].fname;
-
-            // depending on the settings, we'll either add the native unicode as the alt tag, otherwise the shortname
-            const title = mappedUnicode[unicode];
-
-            if (addAlt) {
-                const alt = (emojione.unicodeAlt) ? emojione.convert(unicode.toUpperCase()) : mappedUnicode[unicode];
-                replaceWith = `<img class="mx_emojione" title="${title}" alt="${alt}" src="${emojione.imagePathSVG}${fname}.svg${emojione.cacheBustParam}"/>`;
-            } else {
-                replaceWith = `<img class="mx_emojione" src="${emojione.imagePathSVG}${fname}.svg${emojione.cacheBustParam}"/>`;
-            }
-            return replaceWith;
-        }
-    });
-
-    return str;
-}
-
 /**
  * Returns the shortcode for an emoji character.
  *
  * @param {String} char The emoji character
  * @return {String} The shortcode (such as :thumbup:)
  */
-export function unicodeToShort(char) {
-    const unicode = emojione.jsEscapeMap[char];
-    return emojione.mapUnicodeToShort()[unicode];
+export function unicodeToShortcode(char) {
+    const data = EMOJIBASE.find(e => e.unicode === char);
```

Comment:
Could we include the `unicode` field (and `emoticon` used in the composer) in `stripped-emoji.json` so we don't need to include `emojibase-data/en/compact.json` in the bundle as well? Or if the size difference is negligible, use `compact.json` in `EmojiProvider` as well, and get rid of the stripped script altogether.

Reply:
so, the stripped-emoji.json thing makes zero sense, as per https://github.com/matrix-org/matrix-react-sdk/pull/2995/commits/b0ec594c5a3f2faed4c957d926dd6a65bc17e053. I deliberately didn't try to untangle it though, given it's only used by EmojiProvider and I'm assuming that EmojiProvider is going to die a death when we switch to using emojimart or whatever as a picker.  Or if we end up keeping both the old-school autocomplete as well as emojimart, we fix it then?

Reply:
Ok, makes sense!